cli/src/project.ts

```diff
@@ -3,24 +3,41 @@ import Crate from './crate';
 import Target from './target';
 import BuildSettings from './build-settings';
 import log from './log';
-import { spawn } from './async/child_process';
+import { execFile, spawn } from './async/child_process';
 import * as rust from './rust';
 
 export type ProjectOptions = {
-  crate?: string
+  crate: string
+  targetDirectory: string
 };
 
 /** A Neon project and its directory tree. */
 export default class Project {
   readonly root: string;
+  readonly targetDirectory: string;
   readonly crate: Crate;
 
-  constructor(root: string, options: ProjectOptions = {}) {
-    let { crate = 'native' } = options;
+  private constructor(root: string, options: ProjectOptions) {
+    let { crate, targetDirectory } = options;
     this.root = root;
+    this.targetDirectory = targetDirectory!;
     this.crate = new Crate(this, { subdirectory: crate });
   }
 
+  static async create(root: string, options: Partial<ProjectOptions> = {}): Promise<Project> {
+    let { crate = 'native' } = options;
+    const { stdout } = await execFile("cargo", ["metadata", "--format-version=1"], {
```

Comment:
This is only necessary in `build` and would be fewer changes if it was moved there, but it felt wrong not to have `targetDirectory` on the `Project`.

Reply:
Seems 👍  to me.